third_party/move/move-core/types/src/value.rs

```diff
@@ -195,6 +195,15 @@ impl MoveValue {
     }
 }
 
+// Sporadically checks if the given two input type layouts match
```

Comment:
Is sporadic supposed to be part of the implementation here? Then the name should indicate so, as calling layout_matches on layouts that don't match and not observing an issue could be problematic. 
I suppose not, but if it were the caller's responsibility, then sporadic shouldn't be mentioned anywhere.